### PR TITLE
Ensure test-retry-plugin usages are visible in gradle

### PR DESCRIPTION
**Pull request by jonnybbb**
The internal gradle api usages of the test-retry-plugin should be visible to gradle, so that api changes are considered carefully.

### CODE REVIEW

File: subprojects/base-services/src/main/java/org/gradle/internal/service/ServiceRegistry.java
Comment (jonnybbb) [inline subprojects/base-services/src/main/java/org/gradle/internal/service/ServiceRegistry.java:39]: not sure how we should comment this to avoid any misunderstanding that actuall bot the scan-plugin and the test-retry-plugin are using this api.

File: subprojects/base-annotations/src/main/java/org/gradle/internal/scan/UsedByScanPlugin.java
Comment (marcphilipp) [inline subprojects/base-annotations/src/main/java/org/gradle/internal/scan/UsedByScanPlugin.java:27]:
```suggestion
 * test-retry or the test-distribution plugin clarify their usage in the {@link #value}.
```

File: subprojects/base-services/src/main/java/org/gradle/internal/service/ServiceRegistry.java
Comment (marcphilipp) [reply subprojects/base-services/src/main/java/org/gradle/internal/service/ServiceRegistry.java:39]: When I did this for the test-distribution plugin I left preexisting annotations unchanged.

**Comment by jonnybbb** [comment]
@marcphilipp I think I forgot to merge that one - Is it still ok to merge to release?

File: subprojects/base-services/src/main/java/org/gradle/internal/service/ServiceRegistry.java
Comment (wolfs) [reply subprojects/base-services/src/main/java/org/gradle/internal/service/ServiceRegistry.java:39]: How about `@UsedByScanPlugin("scan, test-retry")`?

**Comment by marcphilipp** [comment]
Someone from the build tool team should approve and decide if `master` or `release` should be used as base branch.

**Comment by jonnybbb** [comment]
@wolfs can you decide if this change should go into release or master.

**Comment by wolfs** [comment]
@jonnybbb I'd say the change should go into `master`, since I don't see any urgency to put it on `release` and the first RC has already been published from `release`.

**Review by wolfs — CHANGES_REQUESTED** [review]
The PR now contains too many changes. Please rebase it on `master`.

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**
